src/protocols/memory/ddr3.stanza

```diff
@@ -0,0 +1,573 @@
+  constructor => #DDR3-Constraint
+
+
+doc: \<DOC>
+Constructor for the DDR3 Link Constraint
+@param ck-rs Differential Routing Structure constraints for CK signals
+@param dqs-rs Differential Routing Structure constraints for DQS signals
+@param dq-rs Routing Structure constraints for DQ signals
+@param rs Routing Structure constraints for remaining signals (CMD/CTRL/ADDR)
+@param data-cst Data Link sub-constraint
+@param acc-cst ACC Link sub-constraint
+@param data-acc-cst Data To ACC sub-constraint
+<DOC>
+public defn DDR3-Constraint (
+    width:DDR3-width,
+    topology:DDR3-topology,
+    rank:DDR3-rank,
+    diff-ck-rs:DifferentialRoutingStructure,
+    diff-dqs-rs:DifferentialRoutingStructure,
+    se-dq-rs:RoutingStructure,
+    se-rs:RoutingStructure,
+    data-cst:DDR3-Data-Constraint,
+    acc-cst:DDR3-ACC-Constraint,
+    data-acc-cst:DDR3-Data-ACC-Constraint ) -> DDR3-Constraint:
+  DDR3-Constraint(
+    width = width,
+    topology = topology,
+    rank = rank,
+    diff-ck-rs = diff-ck-rs,
+    diff-dqs-rs = diff-dqs-rs,
+    se-dq-rs = se-dq-rs,
+    se-rs = se-rs,
+    data-cst = data-cst,
+    acc-cst = acc-cst,
+    data-acc-cst = data-acc-cst
+  )
+
+
+doc: \<DOC>
+Constrain a DDR3 Link assuming that it is a discrete link (not DIMM)
+
+@param cst Constraint Object
+@param src Source End Point - must be of `ddr3-b` type
+@param dst Destination End Point - must be of `ddr3-b` type and match
+the parameterization of `src`, including width.
+<DOC>
+public defmethod constrain (cst:DDR3-Constraint, src:JITXObject, dst:JITXObject) -> False :
+  inside pcb-module:
+
+    val ilane  = width-to-lane(width(cst))
+    val irank  = rank-to-int(rank(cst))
+
+    constrain(data-cst(cst), src.data, dst.data)
+    constrain(acc-cst(cst), src.acc, dst.acc)
+    constrain(data-acc-cst(cst), src, dst)
+
+
+public defn connect-DDR3 (src:JITXObject, dst:JITXObject
+                          -- width:DDR3-width = DDR3-x16,
+                          topology:DDR3-topology = DDR3-FlyBy,
+                          rank:DDR3-rank = DDR3-SingleRank,
+                          diff-ck-rs:DifferentialRoutingStructure,
+                          diff-dqs-rs:DifferentialRoutingStructure,
+                          dq-rs:RoutingStructure,
+                          rs:RoutingStructure) :
+  inside pcb-module :
+    val ddr3-cst = DDR3-Constraint(width = width,
+                                   topology = topology,
+                                   rank = rank,
+                                   diff-ck-rs = diff-ck-rs,
+                                   diff-dqs-rs = diff-dqs-rs,
+                                   se-dq-rs = dq-rs,
+                                   se-rs = rs)
+    constrain-topology(src, dst, ddr3-cst)
+
+doc: \<DOC>
+Constrain a DDR3 Data to ACC Link assuming that it is a discrete link (not DIMM or combined data/control)
+
+@param cst Data to ACC channel constraint object
+@param src Source end point - must be of `ddr3` type
```

Comment:
You use `ddr3-b` above - we should probably be consistent because I think they both take the same.